assets/src/components/v2/bus_shelter/persistentHOC.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { useContext, useEffect, useState } from "react";
```

Comment:
I missed this before, but this should live in the level above the apps. The best place for this is probably `src/apps/components/v2`. I feel like multiple apps would need to use this in the future. Can you also change the name of the file to match the name of the "component"? In this case, `persistent_wrapper.tsx`.